contracts/src/v0.8/dev/special/NoCancelVRFCoordinatorV2.sol

```diff
@@ -0,0 +1,883 @@
+    external
+    override
+    onlySubOwner(subId)
+    nonReentrant
+  {
+    // Proposing to address(0) would never be claimable so don't need to check.
+    if (s_subscriptionConfigs[subId].requestedOwner != newOwner) {
+      s_subscriptionConfigs[subId].requestedOwner = newOwner;
+      emit SubscriptionOwnerTransferRequested(subId, msg.sender, newOwner);
+    }
+  }
+
+  /**
+   * @inheritdoc VRFCoordinatorV2Interface
+   */
+  function acceptSubscriptionOwnerTransfer(uint64 subId) external override nonReentrant {
+    if (s_subscriptionConfigs[subId].owner == address(0)) {
+      revert InvalidSubscription();
+    }
+    if (s_subscriptionConfigs[subId].requestedOwner != msg.sender) {
+      revert MustBeRequestedOwner(s_subscriptionConfigs[subId].requestedOwner);
+    }
+    address oldOwner = s_subscriptionConfigs[subId].owner;
+    s_subscriptionConfigs[subId].owner = msg.sender;
+    s_subscriptionConfigs[subId].requestedOwner = address(0);
+    emit SubscriptionOwnerTransferred(subId, oldOwner, msg.sender);
+  }
+
+  /**
+   * @inheritdoc VRFCoordinatorV2Interface
+   */
+  function removeConsumer(uint64 subId, address consumer) external override onlySubOwner(subId) nonReentrant {
+    if (s_consumers[consumer][subId] == 0) {
+      revert InvalidConsumer(subId, consumer);
+    }
+    // Note bounded by MAX_CONSUMERS
+    address[] memory consumers = s_subscriptionConfigs[subId].consumers;
+    uint256 lastConsumerIndex = consumers.length - 1;
+    for (uint256 i = 0; i < consumers.length; i++) {
+      if (consumers[i] == consumer) {
+        address last = consumers[lastConsumerIndex];
+        // Storage write to preserve last element
+        s_subscriptionConfigs[subId].consumers[i] = last;
+        // Storage remove last element
+        s_subscriptionConfigs[subId].consumers.pop();
+        break;
+      }
+    }
+    delete s_consumers[consumer][subId];
+    emit SubscriptionConsumerRemoved(subId, consumer);
+  }
+
+  /**
+   * @inheritdoc VRFCoordinatorV2Interface
+   */
+  function addConsumer(uint64 subId, address consumer) external override onlySubOwner(subId) nonReentrant {
+    // Already maxed, cannot add any more consumers.
+    if (s_subscriptionConfigs[subId].consumers.length == MAX_CONSUMERS) {
+      revert TooManyConsumers();
+    }
+    if (s_consumers[consumer][subId] != 0) {
+      // Idempotence - do nothing if already added.
+      // Ensures uniqueness in s_subscriptions[subId].consumers.
+      return;
+    }
+    // Initialize the nonce to 1, indicating the consumer is allocated.
+    s_consumers[consumer][subId] = 1;
+    s_subscriptionConfigs[subId].consumers.push(consumer);
+
+    emit SubscriptionConsumerAdded(subId, consumer);
+  }
+
+  /**
+   * @inheritdoc VRFCoordinatorV2Interface
+   */
+  function cancelSubscription(uint64 subId, address to) external override onlySubOwner(subId) nonReentrant {
+    revert("sub cancellation not allowed");
```

Comment:
do we also want to gut cancelSubscriptionOwner?

Reply:
My understanding is that we might still want to retrieve the tokens that are remaining in the subscription, if the need to cancel ever arises.